### PR TITLE
Improve general code

### PR DESCRIPTION
**Pull request by henryboisdequin**
This PR:

1. Improves `malformed_sql_query` test code

### CODE REVIEW

File: diesel/src/pg/connection/mod.rs [hunk 1]
```diff
@@ -193,15 +193,11 @@ mod tests {
...
         let query =
             crate::sql_query("SELECT not_existent FROM also_not_there;").execute(&connection);
+
         if let Err(err) = query {
-            match err {
-                DatabaseError(_, string) => {
-                    assert_eq!(Some(26), string.statement_position());
-                }
-                _ => unreachable!(),
+            if let DatabaseError(_, string) = err {
```
Comment (weiznich) [inline diesel/src/pg/connection/mod.rs:198]: This test should fail if something other than that error is returned. The old variant ensured this, while this one will silently accept other error messages.

File: diesel/src/mysql/types/date_and_time.rs [hunk 2]
```diff
@@ -164,7 +164,7 @@ mod tests {
     use crate::sql_types::{Date, Datetime, Time, Timestamp};
 
     fn connection() -> MysqlConnection {
-        dotenv().ok();
+        let _ = dotenv();
```
Comment (weiznich) [inline diesel/src/mysql/types/date_and_time.rs:167]: I would wait with applying this change till the documentation of `dotenv()` itself recommends this: https://github.com/dotenv-rs/dotenv/pull/57

**Review by weiznich — CHANGES_REQUESTED** [review]

File: diesel/src/mysql/types/date_and_time.rs (hunk 2 above)
Comment (henryboisdequin) [reply diesel/src/mysql/types/date_and_time.rs:167]: @weiznich I've addressed your first comment. Now, I'll remove all the `dotenv()` changes, I will make another PR once that lands.

File: diesel/src/mysql/connection/mod.rs [hunk 3]
```diff
@@ -130,7 +130,7 @@ mod tests {
     use std::env;
 
     fn connection() -> MysqlConnection {
-        let _ = dotenv::dotenv();
```
Comment (henryboisdequin) [inline diesel/src/mysql/connection/mod.rs:133]: Should I keep this? I think it's better to use `.ok()` to keep consistency for now.

File: diesel_cli/src/infer_schema_internals/information_schema.rs [hunk 4]
```diff
@@ -359,7 +359,7 @@ mod tests {
     use std::env;
 
     fn connection() -> PgConnection {
-        let _ = dotenv();
```
Comment (henryboisdequin) [inline diesel_cli/src/infer_schema_internals/information_schema.rs:362]: Same here

**Pull request merged**